Workiva/w-mobile-kit
# MOBILE-1798: iOS status bar appears over modal dialog box

## Description

Whenever the app is displaying a modal with a semi-transparent backdrop, the backdrop should appear on top of the status bar (currently it is the opposite).
## What Was Changed

Changed presentation of action sheets to present a UIWindow with the action sheet on top of that which allows it to appear over the iOS status bar. When the view controller disappears, the window is destroyed, thus the lifecycle is the same as before but appears correctly to the user.

Also added another property on action sheets allowing the handlers to be executed after the action sheet dismisses, thus allowing for smoother transitions when an action sheet handler presents another view controller.
## Testing
- Ensure all unit tests pass
- Ensure new unit tests provide acceptable coverage for new features added
- Ensure for all action sheets that the status bar keeps its preferences based on the presenting view controller (whether it is hidden or not, default or not)
- Ensure the status bar appears under the action sheet properly
- Ensure the lifecycle of the windows matches the lifecycle of the action sheet view controller
  - (One way is to open multiple action sheets in different ways and verify afterwards in the view hierarchy debugger that there are not unused windows sitting in memory)

---

Please Review: @Workiva/mobile
